File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        largest_sum = nums[0]
        
        for num in nums[1:]:
            if max_sum > 0:
                max_sum += num
            else:
                max_sum = num
                
            if max_sum > largest_sum:
                largest_sum = max_sum
                    
        return largest_sum

File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
    currentMax := nums[0]
    globalMax := nums[0]
    
    for i:= 1; i < len(nums); i++ {
        currentMax = max(nums[i], currentMax + nums[i])
        globalMax = max(globalMax, currentMax)
    }
    
    return globalMax
    
}

func max(a, b int) int {
   max := a
   if b > a {
      max = b
   }
   return max
}